src/main/java/net/bramp/ffmpeg/builder/FFmpegHlsOutputBuilder.java

```diff
@@ -19,8 +19,18 @@ public class FFmpegHlsOutputBuilder extends AbstractFFmpegOutputBuilder<FFmpegHl
 
     protected FFmpegHlsOutputBuilder(FFmpegBuilder parent, String filename) {
         super(parent, filename);
+        this.format = "hls";
```

Comment:
I think you've lost the check that the format wasn't already set to a non-hls format. There should either be a check in the constructor, or call setFormat(..)

Reply:
Oh I'm also confused. We have this.format, and parent.format. this will no doing lead to confusion.